DifferentiationInterface/ext/DifferentiationInterfaceHyperHessiansExt/DifferentiationInterfaceHyperHessiansExt.jl

```diff
@@ -0,0 +1,259 @@
+module DifferentiationInterfaceHyperHessiansExt
+
+import DifferentiationInterface as DI
+import .DI: AutoHyperHessians
+using ADTypes: ForwardMode
+using HyperHessians:
+    HVPConfig,
+    HessianConfig,
+    Chunk,
+    chunksize,
+    pickchunksize,
+    hessian,
+    hessian!,
+    hessian_gradient_value,
+    hessian_gradient_value!,
+    hessian,
+    hvp,
+    hvp!,
+    hvp_gradient_value,
+    hvp_gradient_value!
+
+## Traits
+DI.check_available(::DI.AutoHyperHessians) = true
+DI.inplace_support(::DI.AutoHyperHessians) = DI.InPlaceSupported()
+DI.hvp_mode(::DI.AutoHyperHessians) = DI.ForwardOverForward()
+DI.mode(::DI.AutoHyperHessians) = ForwardMode()
+
+chunk_from_backend(backend::DI.AutoHyperHessians, x) =
+    isnothing(backend.chunksize) ? Chunk(x) : Chunk{backend.chunksize}()
+chunk_from_backend(backend::DI.AutoHyperHessians, N::Integer, ::Type{T}) where {T} =
+    isnothing(backend.chunksize) ? Chunk(pickchunksize(N, T), T) : Chunk{backend.chunksize}()
+
+function DI.pick_batchsize(backend::DI.AutoHyperHessians, x::AbstractArray)
+    B = chunksize(chunk_from_backend(backend, x))
+    return DI.BatchSizeSettings{B}(length(x))
+end
+
+## Second derivative (scalar input)
+
+struct HyperHessiansSecondDerivativePrep{SIG} <: DI.SecondDerivativePrep{SIG}
+    _sig::Val{SIG}
+end
+
+function DI.prepare_second_derivative_nokwarg(
+        strict::Val, f, backend::DI.AutoHyperHessians, x::Number, contexts::Vararg{DI.Context, C}
+    ) where {C}
+    _sig = DI.signature(f, backend, x, contexts...; strict)
+    return HyperHessiansSecondDerivativePrep(_sig)
+end
+
+function DI.second_derivative(
+        f,
+        prep::HyperHessiansSecondDerivativePrep,
+        backend::DI.AutoHyperHessians,
+        x::Number,
+        contexts::Vararg{DI.Context, C},
+    ) where {C}
+    DI.check_prep(f, prep, backend, x, contexts...)
+    fc = DI.fix_tail(f, map(DI.unwrap, contexts)...)
```

Comment:
This will only work with `Constant` contexts (see the ForwardDiff extension for how I deal with `Cache`)